pkgs/applications/emulators/dosbox-x/default.nix

```diff
@@ -0,0 +1,87 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, alsa-lib
+, autoconf
+, automake
+, file
+, fluidsynth
+, freetype
+, glib
+, libpcap
+, libpng
+, libslirp
+, libXext
+, libxkbfile
+, libXrandr
+, nasm
+, ncurses
+, pkg-config
+, SDL2
+, which
+, yad
+, zlib
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dosbox-x";
+  version = "2023.09.01";
+
+  src = fetchFromGitHub {
+    owner = "joncampbell123";
+    repo = "dosbox-x";
+    rev = "dosbox-x-v${finalAttrs.version}";
+    hash = "sha256-gEqIPXrq16p6VC9aGW2sRSWE0SG9hiMWeRzSoDvUbB0=";
+  };
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    file
+    pkg-config
+  ];
+
+  buildInputs = [
+    alsa-lib
+    fluidsynth
+    freetype
+    glib
+    libpcap
+    libpng
+    libslirp
+    libXext
+    libxkbfile
+    libXrandr
+    nasm
```

Comment:
If this is supposed to be used at build time, then it should be in `nativeBuildInputs`. `buildInputs` isn't supposed to make binaries available at build time - setting `strictDeps = true;` will enforce this separation the same way cross-compiling would (I would recommend this in general to enforce good practices).

But, is NASM *actually* being used in our case? I can only really find references to it in the vendored SDL1 which we don't use, and a vendored FLAC library where I'm not sure if that is being used for anything. Could you try to dig around and see if this is actually used for anything in the build we'd be trying to do?

Reply:
```
checking for nasm... no
configure: WARNING: NASM (netwide assembler) not found, you will not be able to compile the external x86 .asm files if they are modified
```

I think it's perhaps more useful for development but I'm open to omitting it from the build inputs if you prefer. Definitely can move to nativebuildinputs.

Reply:
I see that it's checking for NASM and warning when it's not found, but if it's just for manually compiling optional stuff for the devs (for example the copy of MASTER.LIB in ref), and it's not used in the actual build, then no point in pulling it in. If someone wants to assemble examples or smth, then just `nix-shell -p nasm` & `nix-build -A dosbox-x.src` or `git clone`.

Reply:
sure, I'll remove it.